applications/tari_app_utilities/src/identity_management.rs

```diff
@@ -147,6 +147,33 @@ pub fn create_new_identity<P: AsRef<Path>>(
     Ok(node_identity)
 }
 
+/// Recover a node id from a given private key and save it to disk
+/// ## Parameters
+/// `private_key` - The private key
+/// `path` - Reference to path to save the file
+/// `public_addr` - Network address of the base node
+/// `peer_features` - The features enabled for the base node
+///
+/// ## Returns
+/// A NodeIdentity wrapped in an atomic reference counter on success, the exit code indicating the reason on failure
+pub fn recover_node_identity<P: AsRef<Path>>(
```

Comment:
This isn't actually used, I thought it was necessary but it's not required. Figured I'd leave it in as it might be useful? Happy to remove this function @mikethetike